src/main/java/ir/nft/security/oauth2manager/controller/OAuth2GroupsController.java

```diff
@@ -0,0 +1,33 @@
+package ir.nft.security.oauth2manager.controller;
+
+import ir.nft.security.oauth2manager.dto.domain.OAuth2GroupDTO;
+import ir.nft.security.oauth2manager.dto.idcombination.OAuth2GroupIdDTO;
+import ir.nft.security.oauth2manager.service.OAuth2GroupService;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(APISpecifications.GROUPS_API_V1_URL)
+public class OAuth2GroupsController {
+
+  private final OAuth2GroupService groupService;
+
+  public OAuth2GroupsController(OAuth2GroupService groupService) {
+    this.groupService = groupService;
+  }
+
+  @PostMapping
```

Comment:
this method and `getGroupById` have same path?

Reply:
Yes It is. it's from the rest conventions that they add new entitties by posting to the root of the corresponding api and they don't mention the functionality of that request in the url. except for users that /login and /register are commonly used. how would you like the difference to be? using sth like /add or /get?